File: 0009-palindrome-number/0009-palindrome-number.cs
public class Solution {
    public bool IsPalindrome(int x) {
        if(x < 0) // signed values aren't palindromes
            return false;
        
        int num = x; // copy of x for calculation
        int rev = 0; // reversed value of num
        
        while(num != 0) 
        {
            rev = (rev * 10) + (num % 10);
            num /= 10;
        }
        
        return x == rev;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.php
class Solution {

    /**
     * @param Integer $x
     * @return Boolean
     */
    function isPalindrome($x) {
        if($x < 0) // signed values aren't palindromes
            return false;
        
        $num = $x; // copy of $x for calculation
        $rev = 0; // reversed value of $num
        
        while($num != 0) {
            $rev = ($rev * 10) + ($num % 10);
            $num = (int)($num / 10); // integer division
        }
        
        return $x == $rev;
    }
}